MapboxNavigation/RouteMapViewController.swift

```diff
@@ -253,6 +262,51 @@ extension RouteMapViewController: NavigationMapViewDelegate {
         let route = routeController.routeProgress.route
         guard let coordinates = route.coordinates else  { return nil }
         
+        if let userLocation = mapView.userLocation,
+            let style = mapView.style,
+            routeController.showCurrentWayNameLabel,
+            recenterButton.isHidden {
+            
+            let streetsSources = style.sources.flatMap {
+                $0 as? MGLVectorSource
+                }.filter {
+                    $0.isMapboxStreets
+            }
+            let streetsSourceIdentifiers = streetsSources.map {
+                $0.identifier
+            }
+            assert(!streetsSourceIdentifiers.isEmpty, "The option `showCurrentWayNameLabel` must contain the source `mapbox.mapbox-streets-v7`")
+            
+            if let mapboxSteetsSource = streetsSources.first, style.layer(withIdentifier: "roadLabelLayer") == nil{
+                let streetLabelLayer = MGLLineStyleLayer(identifier: RoadLabelLayerIdentifier, source: mapboxSteetsSource)
+                streetLabelLayer.sourceLayerIdentifier = "road_label"
+                
+                // If the opacity is set to 0, the feature will be ignored in `mapView.visibleFeatures()`
+                streetLabelLayer.lineOpacity = MGLStyleValue(rawValue: 0.001)
+                streetLabelLayer.lineWidth = MGLStyleValue(rawValue: 10)
+                style.addLayer(streetLabelLayer)
+            }
+            
+            if let userPuck = mapView.view(for: userLocation) {
+                let features = mapView.visibleFeatures(in: userPuck.frame, styleLayerIdentifiers: Set([RoadLabelLayerIdentifier]))
+                
+                for feature in features {
+                    
+                    var key = "name"
+                    if let language = Locale.preferredLanguages.first!.components(separatedBy: "-").first,
```

Comment:
It’d be great to adapt this UI to the user’s preferred language, but are we doing that in the turn banner and step table and on the map? I think it’d be more important to be consistent throughout the UI. We could use the runtime styling API to localize the map, with code very similar to this code. However, the Directions API doesn’t yet return localized way names, so the turn banner and step table are stuck in the local language.